src/main/java/me/crespel/runtastic/converter/ExportConverter.java

```diff
@@ -9,60 +9,125 @@
  */
 public class ExportConverter {
 
 	public static final String SPORT_SESSIONS_DIR = "Sport-sessions";
+	public static final String PHOTOS_DIR = "Photos";
+	public static final String PHOTOS_META_DATA_DIR = "Photos" + File.separator + "Images-meta-data";
+	public static final String PHOTOS_SPORT_SESSION_ALBUMS_DIR = "Photos" + File.separator + "Images-meta-data" + File.separator + "Sport-session-albums";
+	public static final String USER_DIR = "User";
 	public static final String DEFAULT_FORMAT = "tcx";
 
 	protected final SportSessionParser parser = new SportSessionParser();
 	protected final SportSessionMapper<?> mapper = new DelegatingSportSessionMapper();
 
-	public List<SportSession> listSportSessions(File path) throws FileNotFoundException, IOException {
-		path = normalizeSportSessionPath(path);
+	public List<SportSession> listSportSessions(File path, boolean full) throws FileNotFoundException, IOException {
+		path = normalizeExportPath(path, SPORT_SESSIONS_DIR);
 		List<SportSession> sessions = new ArrayList<>();
 		File[] files = path.listFiles(file -> file.getName().endsWith(".json"));
 		for (File file : files) {
-			sessions.add(parser.parseSportSession(file));
+			sessions.add(parser.parseSportSession(file,full));
 		}
 
 		Collections.sort(sessions);
 		return sessions;
 	}
 
+	public User getUser(File path) throws FileNotFoundException, IOException {
+		return parser.parseUser(new File(normalizeExportPath(path, USER_DIR), "user.json"));
+	}
+
 	public SportSession getSportSession(File path, String id) throws FileNotFoundException, IOException {
-		path = normalizeSportSessionPath(path);
-		return parser.parseSportSession(new File(path, id + ".json"));
+		return parser.parseSportSession(new File(normalizeExportPath(path, SPORT_SESSIONS_DIR), id + ".json"), true);
+	}
+
+	public SportSession getSportSessionWithPhoto(File path, String photoid) throws FileNotFoundException, IOException {
+		String sessionid = null;
+
+		File photofile = new File(normalizeExportPath(path, PHOTOS_DIR), photoid + ".jpg");
+		if( photofile.exists() ) {
+			// photo file found ...
+
+			ImagesMetaData image = parser.parseImagesMetaData(new File(normalizeExportPath(path, PHOTOS_META_DATA_DIR), photoid + ".json"));
+			if( image != null ) {
+				// photo meta data file found ...
+
+				// search trough sport session album data, to find sport session related to the photo
+				File[] files = normalizeExportPath(path, PHOTOS_SPORT_SESSION_ALBUMS_DIR).listFiles(file -> file.getName().endsWith(".json"));
+				for( File file : files ) {
+					try {
+						SportSessionAlbums mysessionalbum = parser.parseSportSessionAlbumsData(file);
+						if (mysessionalbum.getPhotosIds().contains(photoid)) {
+							// Sport session id found within sport session albums
+							sessionid = mysessionalbum.getId();
+						}
+					} catch (IOException e) {
+						throw new RuntimeException(e);
+					}
+				}
+			}
+		}
+
+		return getSportSession(path, sessionid);

```

Comment:
If the photofile does not exist, this method should not try to `getSportSession()` with a null sessionid, it should throw a FileNotFoundException instead.

Reply:
Good hint, I check now session id and throw an execption:

if( sessionid == null) {
	throw new FileNotFoundException("Sport Session file not found");
}